.github/workflows/render-task-definition.yml

```
@@ -97,26 +112,55 @@ jobs:
         with:
           ref: ${{ inputs.ref }}
 
+      - name: 'Setup python'
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ inputs.python-version }}
+          python-version-file: ${{ inputs.python-version-file }}
+          cache: pip
+
+      - name: 'If ssh private key provided, setup ssh agent'
+        id: ssh
+        run: |
+          has_key='false'
+          [ -z "${{ secrets.ssh-private-key }}" ] || has_key='true'
+          echo "has_key=$has_key" >> "$GITHUB_OUTPUT"
+
+      # for private dependencies
+      - name: 'Install SSH Key'
+        if: steps.ssh.outputs.has_key == 'true'
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.ssh-private-key }}
+
+      - name: 'Install python dependencies'
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          if [ -n "${{ inputs.requirements-file }}" ]; then
+            pip install -r "${{ inputs.requirements-file }}"
+          elif [ -f pyproject.toml ]; then
+            pip install .
+          fi
+
       - name: 'Render task definition'
-        uses: shopsmart/render-j2-action@v2
+        id: render
+        uses: shopsmart/github-actions/actions/render-jinjanator-template@refactor/render-task-definition
```

Since there were no changes to this workflow in the current v3, I am still going to release this under the v3 tag.  All of our actions referencing this workflow are using the v2 tag.

```suggestion
        uses: shopsmart/github-actions/actions/render-jinjanator-template@v3
```